octodns_constellix/__init__.py

```diff
@@ -861,33 +999,58 @@ def _gen_pool_data(self, record):
             }
         return pool_data
 
+    def _gc_healthchecks(self, checks):
+        for check in checks:
+            self._delete_check(check['type'], check['name'])
+
     def _create_update_dynamic_healthchecks(self, record, pool_data):
+        self.log.debug(
+            '_create_update_dynamic_healthchecks %r %r', record.data, pool_data
+        )
         healthcheck = self._healthcheck_config(record)
-        if healthcheck is None:
-            return pool_data, None
-
         check_sites = self._sonar.agents_for_regions(
-            healthcheck['sonar_regions']
+            healthcheck['regions'], healthcheck['site_limit']
         )
+        check_fqdn = healthcheck.get('host', None)
+        check_path = healthcheck.get('path', None)
+
         health_data = {}
+        deleted_checks = []
         for pool_name, pool in pool_data.items():
             for value in pool['values']:
+                policy = value.get('policy', 'followsonar')
+
+                check_type = healthcheck['type'].lower()
                 check_name = '{}-{}'.format(pool_name, value['value'])
-                check_obj = self._create_update_check(
-                    pool_type=record._type,
-                    check_name=check_name,
-                    check_type=healthcheck['sonar_type'].lower(),
-                    value=value['value'],
-                    port=healthcheck['sonar_port'],
-                    interval=healthcheck['sonar_interval'],
-                    sites=check_sites,
+                self.log.debug(
+                    'update policy %s %s %s', policy, check_type, check_name
                 )
-                value['checkId'] = check_obj['id']
-                value['policy'] = 'followsonar'
-                health_data[check_name] = check_obj
-        return pool_data, health_data
+                if policy == 'followsonar':
+                    check_obj = self._create_update_check(
+                        pool_type=record._type,
+                        check_name=check_name,
+                        check_type=check_type,
+                        value=value['value'],
+                        port=healthcheck['port'],
+                        fqdn=check_fqdn,
+                        path=check_path,
+                        interval=healthcheck['interval'],
+                        policy='ONCEPERREGION',
```

Comment:
@ross @viranch this option is necessary to keep the costs down.

Without this, a _single_ WORLD monitor would cost about USD 185 / month. With this option, it costs about USD 3 / month.

We could add a config flag to change this later on.

Reply:
So long as it's documented, ideally with links to their info about the policies. It should be fine and as you say we can always add `octodns.constellix.<whatever>` options if needed.